### PR TITLE
Add ClickHouse as a supported datasource

### LINKED ISSUE #18: Use clickhouse instead of Mongo as backend storage?
Could you think about using ClickHouse instead of MongoDB?
Please look to http://clickhouse.yandex/docs/en/
ClickHouse is a blazing fast Columnar database, which contains scale-out cluster data models

### PR DESCRIPTION
**Pull request by jdorn**
Fixes #18

### CODE REVIEW

File: packages/back-end/src/integrations/ClickHouse.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      .replace("T", " ")}')`;
+  }
+  addDateInterval(col: string, days: number) {
+    return `date_add(day, ${days}, ${col})`;
+  }
+  subtractHalfHour(col: string) {
+    return `date_sub(hour, 30, ${col})`;
+  }
+  regexMatch(col: string, regex: string) {
+    // Does the regex need to escape `\` here?
```
Comment (Slach) [inline packages/back-end/src/integrations/ClickHouse.ts:52]: re2 regexp use here and yes, back slash required for escaping look to 
https://clickhouse.tech/docs/en/sql-reference/functions/string-search-functions/#matchhaystack-pattern

File: packages/front-end/components/Settings/ClickHouseForm.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  existing: boolean;
+  onParamChange: ChangeEventHandler<HTMLInputElement>;
+}> = ({ params, existing, onParamChange }) => {
+  return (
+    <>
+      {isCloud() ? (
+        <div className="row">
+          <div className="col-auto">
+            <div className="alert alert-info">
+              Make sure to whitelist the IP Address <code>52.70.79.40</code> so
```
Comment (Slach) [inline packages/front-end/components/Settings/ClickHouseForm.tsx:16]: hmm, could growbook used as self-hosted solution?

File: packages/back-end/src/integrations/ClickHouse.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    return `match(${col}, "${regex}")`;
```
Comment (Slach) [inline packages/back-end/src/integrations/ClickHouse.ts:53]: also please use single quotes for regexp `'${regexp}'`
double quotes mean "field_name" in ClickHouse

Comment (jdorn) [reply packages/back-end/src/integrations/ClickHouse.ts:53]: Sorry, saw this after merging.  Just fixed in a followup commit.

**Pull request merged**